third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -0,0 +1,512 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Prerequisites: none.
+//!
+//! Simplifies the control flow graph in the following ways:
+//! (Implemented by `ControlFlowGraphSimplifier`)
+//! - eliminates branch/jump to jump
+//!     L1: goto L2 (L1 != L2)
+//!     =>
+//!     (removed)
+//!
+//!     goto L1
+//!     =>
+//!     goto L2
+//!
+//!     if ... goto L1 else ... / if ... goto ... else L1
+//!     =>
+//!     if ... goto L2 else ... / if ... goto ... else L2
+//! - removes edges where the source has only one successor and the target has only one predecessor
+//!     BB1: xx
+//!          goto L2
+//!     // BB1 BB2 don't have to be consecutive
+//!     BB2: L2 (no other goto L2)
+//!          yy
+//!     =>
+//!     BB1: xx
+//!          yy
+//! - removes all jumps to the next instruction:
```

Comment:
?

Reply:
The comment is outdated now. But it asked if rule 3 is a special case of rule 2. It's not, since rule 3 doesn't require the block with Label L has only 1 predecessor.